src/components/inputs/single/SingleJourney.tsx

```diff
@@ -1,44 +1,86 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Calender from '../../shared/Input/Calendar';
 import AutoComplete from '../../shared/Input/AutoComplete';
 import Timer from '../../shared/Input/Timer';
 import { Button as PRButton } from 'primereact/button';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { InputActionTypes } from '../../../redux/actions';
+import { IInputs } from '../../../lib/models/inputs';
+import { getInputs } from '../../../redux/selectors/inputs';
 
 function SingleJourney() {
     const dispatch = useDispatch();
+    const defaultInputs = useSelector(getInputs)
+    const [userSource, setUserSource] = useState<string>(defaultInputs.source.text);
```

Comment:
In any case where you think state might be complex but of similar kind, like here, state can be an object and it is similar kind (that is form inputs), we can use useReducer. Here we had only 4 setStates, but yeah if you think it is going to be more than that, you can use useReducer.